Expensify/App
# [No QA] Build Cherry-pick flow for Expensify.cash

**Note:** This PR creates a CP-staging workflow _only_. Not CP-production.

### Details
Creates the ability to cherry-pick a workflow to staging by merging a pull request with the `CP Staging` label.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/155229

### Testing plan
1. Ensure that the `StagingDeployCash` is _not_ locked
1. Create a pull request with the `CP Staging` label. `OSBotify` should comment on the pull request with a comment that looks like:

    ![image](https://user-images.githubusercontent.com/47436092/119580901-4072fb80-bd76-11eb-8724-fec4802abece.png)

1. Merge the pull request with the `CP Staging` label. It should follow the normal process and be deployed to staging.
1. Lock the `StagingDeployCash`. The normal chain-of-events should happen, culminating with a staging deploy.  The `cherryPick.yml` workflow should _not_ be executed.
1. Merge a pull request with an easily identifiable change (no `CP Staging` label). Call this PR A. As per usual, it _should not_ be deployed or added to the `StagingDeployCash`. A comment should appear on the PR stating that it will be deployed later.
1. @Jag96 - Merge a pull request with an easily identifiable change with the `CP Staging` label on the PR. Call this PR B. A new version should be created and merged to main, and the `preDeploy` workflow should synchronously execute the `cherryPick.yml` workflow.
1. A staging deploy should occur, and a comment should appear on PR B stating that it was _cherry picked_ (not just deployed) to staging in the correct version.
1. Once the staging deploy completes, verify that PR A is not present on staging, but PR B is. This is where the easily-identifiable changes will come in handy.
1. Verify that PR B was added to the `StagingDeployCash`, but PR A was not.
1. @Jag96 - Use the GitHub UI to attempt to manually CP PR A to staging. The workflow should not complete. Verify that no staging deploy happens.
1. @AndrewGable or @roryabraham - Use the GitHub UI to attempt to manually CP PR A to staging. It should work and a staging deploy should occur.
1. Once the staging deploy completes, verify that PR A is present on staging.
1. Verify that PR A was added to the `StagingDeployCash`.
1. Unlock the `StagingDeployCash` and revert any temporary changes.
